### PR TITLE
Write integration tests

### PR DESCRIPTION
**Pull request by emmiiorji**
This change adds test for the index and show views of posts and users.

### CODE REVIEW

File: spec/features/posts/index.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      expect(comment_and_like).to have_content('Likes: ')
+      expect(comment_and_like).to have_content(@post1.likes_counter.to_s)
+    end
+  end
+
+  it "redirects to that post's show page after clicking a post" do
+    find("a[href='#{user_post_path(@user.id, @post1.id)}']").click
+    expect(page).to have_selector("input[type='submit'][value='Like']")
+  end
+end
```
Comment (Nemwel-Boniface) [inline spec/features/posts/index.html.erb_spec.rb:59]: - [ ] For some reason, when I run rspec to check whether the specs run or not, I am getting this error. Kindly look into why this is the case.

image

File: spec/features/posts/show.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    end
+  end
+
+  it 'shows comments of each commentors' do
+    comments = @post.comments
+    comments.each do |comment|
+      expect(page).to have_content(comment.text)
+    end
+  end
+end
```
Comment (Nemwel-Boniface) [inline spec/features/posts/show.html.erb_spec.rb:44]: - [ ] For some reason, when I run rspec to check whether the specs run or not, I am getting this error. Kindly look into why this is the case.

image

File: spec/features/users/index.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  end
+  it "When I click on a user, I am redirected to that user's show page" do
+    find(@user1_url).click
+    expect(page).to have_content(@user1.bio)
+  end
+  it "When I click on a user, I am redirected to that user's show page" do
+    find(@user2_url).click
+    expect(page).to have_content(@user2.bio)
+  end
+end
```
Comment (Nemwel-Boniface) [inline spec/features/users/index.html.erb_spec.rb:34]: - [ ] For some reason, when I run rspec to check whether the specs run or not, I am getting this error. Kindly look into why this is the case.

image

File: spec/features/users/show.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  it 'has the see all posts button' do
+    button = find("#{@posts_url} button")
+    expect(button).to have_text('See all posts')
+  end
+
+  it 'redirects to posts index on clicking of the See all posts button' do
+    find(@posts_url).click
+    new_post_anchor = find("a[href='#{new_user_post_path(@user.id)}']")
+    expect(new_post_anchor).to have_content('New Post')
+  end
```
Comment (Nemwel-Boniface) [inline spec/features/users/show.html.erb_spec.rb]: - [ ] For some reason, when I run rspec to check whether the specs run or not, I am getting this error. Kindly look into why this is the case.

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @emmiiorji       :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/users/show.html.erb_spec.rb (hunk 4 above)
Comment (emmiiorji) [reply spec/features/users/show.html.erb_spec.rb]: Hi @Nemwel-Boniface

Thank you for your kind review. We seem not to be getting these errors from our end.

image

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @emmiiorji 

Your project is complete! Great job🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request✔️ 
- Capybara gem ✔️ 
- No linter errors✔️ 

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**